src/storage/txn/scheduler.rs

```diff
@@ -627,31 +628,42 @@ impl<E: Engine, L: LockManager> Scheduler<E, L> {
                 } else if to_be_write.modifies.is_empty() {
                     scheduler.on_write_finished(cid, pr, Ok(()), lock_guards, false, tag);
                 } else {
-                    let sched = scheduler.clone();
-                    // The normal write process is respond to clients and release latches
-                    // after async write finished. If pipelined pessimistic lock is enabled,
-                    // the process becomes parallel and there are two msgs for one command:
-                    //   1. Msg::PipelinedWrite: respond to clients
-                    //   2. Msg::WriteFinished: deque context and release latches
-                    // The order between these two msgs is uncertain due to thread scheduling
-                    // so we clone the result for each msg.
-                    let (write_finished_pr, pipelined_write_pr) = if pipelined {
-                        (pr.maybe_clone().unwrap(), pr)
+                    let proposed_cb: Option<ExtCallback> = if pipelined {
+                        // The normal write process is respond to clients and release latches
+                        // after async write finished. If pipelined pessimistic locking is enabled,
+                        // the process becomes parallel and there are two msgs for one command:
+                        //   1. Msg::PipelinedWrite: respond to clients
+                        //   2. Msg::WriteFinished: deque context and release latches
+                        // The order between these two msgs is uncertain due to thread scheduling
+                        // so we clone the result for each msg.
+                        let pipelined_write_pr = pr.maybe_clone().unwrap();
```

Comment:
Can the result be taken instead of cloned?

Reply:
Do you mean putting it in the `ctx`?

Reply:
Perhaps. I didn't find other way to do this.
I'm actually trying to allow early returning for async cmmit prewrites, but I find it hard to clone the PrewriteResult.

Reply:
But it will need an additional mutex locking to put it in the ctx. I'm not sure if it worth a try.

Reply:
I'm thinking whether it's needed to invoke these callbacks in the scheduler pool because they are simple. I think they won't block raftstore threads.